icu4c/source/common/ucnvmbcs.cpp

```diff
@@ -540,12 +540,12 @@ const UConverterSharedData _MBCSData={
 static const uint32_t
 gb18030Ranges[14][4]={
     {0x10000, 0x10FFFF, LINEAR(0x90308130), LINEAR(0xE3329A35)},
-    {0x9FA6, 0xD7FF, LINEAR(0x82358F33), LINEAR(0x8336C738)},
+    {0x9FBC, 0xD7FF, LINEAR(0x82359135), LINEAR(0x8336C738)}, // for -2022 replace beginning of range with explicit mappings
     {0x0452, 0x1E3E, LINEAR(0x8130D330), LINEAR(0x8135F436)},
     {0x1E40, 0x200F, LINEAR(0x8135F438), LINEAR(0x8136A531)},
     {0xE865, 0xF92B, LINEAR(0x8336D030), LINEAR(0x84308534)},
     {0x2643, 0x2E80, LINEAR(0x8137A839), LINEAR(0x8138FD38)},
-    {0xFA2A, 0xFE2F, LINEAR(0x84309C38), LINEAR(0x84318537)},
+    {0xFA2A, 0xFE0F, LINEAR(0x84309C38), LINEAR(0x84318235)}, // for -2022 replace end of range with explicit mappings
```

Comment:
how about also adding
```
+ {0xFE1A, 0xFE2F, LINEAR(0x84318336), LINEAR(0x84318537)},
```
and not adding
```
<UFE1A> \x84\x31\x83\x36 |0
<UFE1B> \x84\x31\x83\x37 |0
<UFE1C> \x84\x31\x83\x38 |0
<UFE1D> \x84\x31\x83\x39 |0
<UFE1E> \x84\x31\x84\x30 |0
<UFE1F> \x84\x31\x84\x31 |0
<UFE20> \x84\x31\x84\x32 |0
<UFE21> \x84\x31\x84\x33 |0
<UFE22> \x84\x31\x84\x34 |0
<UFE23> \x84\x31\x84\x35 |0
<UFE24> \x84\x31\x84\x36 |0
<UFE25> \x84\x31\x84\x37 |0
<UFE26> \x84\x31\x84\x38 |0
<UFE27> \x84\x31\x84\x39 |0
<UFE28> \x84\x31\x85\x30 |0
<UFE29> \x84\x31\x85\x31 |0
<UFE2A> \x84\x31\x85\x32 |0
<UFE2B> \x84\x31\x85\x33 |0
<UFE2C> \x84\x31\x85\x34 |0
<UFE2D> \x84\x31\x85\x35 |0
<UFE2E> \x84\x31\x85\x36 |0
<UFE2F> \x84\x31\x85\x37 |0
```
to  icu4c/source/data/mappings/gb18030.ucm

Reply:
Same, I think that adding 2-030 explicit mappings has less perf impact than adding another range (which is always checked before any explicit lookup)

Reply:
sure, but there are binary size impact.

Reply:
Please do not change any of these ranges.

Reply:
My mistake, I thought the algorithmic ranges were checked first before the explicit mappings. Restored the original ranges.